inductiva/tasks/run_simulation.py

```diff
@@ -39,4 +45,26 @@ def run_simulation(
         raise RuntimeError(
             f"Expected result to be a string with task_id, got {type(task_id)}")
 
+    if not format_utils.getenv_bool("DISABLE_TASK_METADATA_LOGGING", False):
+        metadata = {
+            **{
+                "task_id": task_id,
+                "input_dir": str(input_dir),
+                "storage_dir": str(storage_dir),
+                "api_method_name": api_method_name,
+                "machine_group_id": resource_pool_id,
+            },
+            **kwargs,
+        }
+        _save_metadata(metadata)
+
     return task
+
+
+def _save_metadata(metadata):
+    """Appends metadata to the TASK_METADATA_FILENAME in the cwd."""
+    file_path = pathlib.Path().cwd() / TASK_METADATA_FILENAME
```

Comment:
Here you should use the `inductiva.utils.files.resolve_path` which does this and/or sets in the desired `inductiva.working_dir` established by the user.